src/s2ir.c

```diff
@@ -765,10 +766,8 @@ class S2irVisitor : public Visitor
                  */
                 if (s->exp->op == TOKstructliteral)
                 {
-                    StructLiteralExp *se = (StructLiteralExp *)s->exp;
-                    se->sym = irs->shidden;
-                    se->fillHoles = true;
-                    e = toElemDtor(s->exp, irs);
```

Comment:
OK, I stopped to use `toElemStructLit` directly in here, and add a test case which was broken by previous *bad* refactoring.